my-webxr-app/src/components/DataPoint.tsx

```diff
@@ -43,17 +46,37 @@ export default function DataPoint({
     setHoverCount(amount);
   };
 
+  // Display the point characteristics in console
+  const showSelectedDataPointInfo = (id: number, marker: string, color: string, column1: string, column2: string, column3: number, meshProps?: MeshProps ) => {
+    console.log("Selected Data Point:");
+    console.log("ID:", id);
+    console.log("Marker:", marker);
+    console.log("Color:", color);
+    console.log("Position:", meshProps?.position);
+    console.log("Column 1:", column1);
+    console.log("Column 2:", column2);
+    console.log("Column 3:", column3);
+  }
+
   return (
     <Interactive
       onHover={() => adjustHoverCount(hoverCount + 1)}
       onBlur={() => adjustHoverCount(hoverCount - 1)}
       onSelect={() => {
-        selectedDataPoint === id
-          ? setSelectedDataPoint(null)
-          : setSelectedDataPoint(id);
+        // Clear current selection
+        if (selectedDataPoint?.id === id) {
```

Comment:
is it necessary to clear `selectedDataPoint` before setting it to a new value?
or can we just do only 1 step of setting it the a new value?

Reply:
Yes it is required to clear selectedDataPoint if the user wants to de-select the current point selection. 

If we only set it to the new value and then select another point, the previously selected point fields will still display in the space. I guess this is something to think about when we implement the multiple point selection requirement.